ext/session/session.c

```diff
@@ -1531,36 +1542,50 @@ PHPAPI void php_session_start(void) /* {{{ */
 		PS(id) = NULL;
```

Comment:
@KalleZ Thank you for checking!
php_session_start() only initialize session ID from client. Invalid session ID or initial access can results in PS(id) == NULL.

There is php_session_initialize() bellow. It creates new session ID when PS(id) == NULL. If it failed for reasons, it should return FAILURE with PS(id) == NULL or PS(id) != NULL.